spec/unit/info_service_spec.rb

```diff
@@ -0,0 +1,236 @@
+                 {:name=>"foo2_b", :type=>"Integer"},
+                 {:name=>"foo2_c", :type=>"String", :default_literal=>"c default value"}
+               ]
+             }
+           ],
+          "#{code_dir}/bar.pp"=>[
+            {:name=>"bar",
+              :params=>[
+                {:name=>"bar_a"},
+                {:name=>"bar_b", :type=>"Integer"},
+                {:name=>"bar_c", :type=>"String", :default_literal=>"c default value"}
+              ]},
+            {:name=>"bar2",
+              :params=>[
+                {:name=>"bar2_a"},
+                {:name=>"bar2_b", :type=>"Integer"},
+                {:name=>"bar2_c", :type=>"String", :default_literal=>"c default value"}
+              ]
+            }
+          ],
+
+          } # end production env
+        }
+      )
+    end
+
+    it "produces classes and parameters from multiple files in multiple environments" do
+      files_production = ['foo.pp', 'bar.pp'].map {|f| File.join(code_dir, f) }
+      files_test = ['fee.pp', 'fum.pp'].map {|f| File.join(code_dir, f) }
+      result = Puppet::InfoService.classes_per_environment({
+        'production' => files_production,
+        'test'       => files_test
+      })
+
+      expect(result).to eq({
+        "production"=>{
+           "#{code_dir}/foo.pp"=>[
+             {:name=>"foo",
+               :params=>[
+                 {:name=>"foo_a"},
+                 {:name=>"foo_b", :type=>"Integer"},
+                 {:name=>"foo_c", :type=>"String", :default_literal=>"c default value"}
+               ]},
+             {:name=>"foo2",
+               :params=>[
+                 {:name=>"foo2_a"},
+                 {:name=>"foo2_b", :type=>"Integer"},
+                 {:name=>"foo2_c", :type=>"String", :default_literal=>"c default value"}
+               ]
+             }
+           ],
+          "#{code_dir}/bar.pp"=>[
+            {:name=>"bar",
+              :params=>[
+                {:name=>"bar_a"},
+                {:name=>"bar_b", :type=>"Integer"},
+                {:name=>"bar_c", :type=>"String", :default_literal=>"c default value"}
+              ]},
+            {:name=>"bar2",
+              :params=>[
+                {:name=>"bar2_a"},
+                {:name=>"bar2_b", :type=>"Integer"},
+                {:name=>"bar2_c", :type=>"String", :default_literal=>"c default value"}
+                ]
+              }
+            ],
+          }, # end production env
+        "test"=>{
+           "#{code_dir}/fee.pp"=>[
+             {:name=>"fee",
+               :params=>[
+                 {:name=>"fee_a", :type=>"Integer", :default_source=>"1+1"}
+               ]},
+           ],
+          "#{code_dir}/fum.pp"=>[
+            {:name=>"fum",
+              :params=>[
+                {:name=>"fum_a"}
+              ]},
+          ],
+         } # end test env
+        }
+      )
+    end
+
+    it "avoids parsing file more than once when environments have same feature flag set" do
```

Comment:
yes, in the future there can be one or more features (we anticipate at least one in the future, but there may be more).